### ISSUE TITLE
Does RHO select once within a epoch or select within each step? 

### ISSUE THREAD

**Issue by xo28** [OATML/RHO-Loss#6]
Hello authors! Thank you for this great work!

I'm new to pytorch-lightning and trying to run/modify some of RHO-LOSS's code. In the paper, Algorithm 1 says:
```
for l = 0,1,... do
    Randomly select a large batch B_t of size n_B
```
In my understanding, I think it means RHO only randomly chooses one batch (w/ batchsize=320 for cifar10) and selects the top 10% of this batch for each epoch. In other words, training data numbers are 10% * batchsize within an epoch.

But I found the training_step() function in MultiModels.py. It receives a batch and makes the selection for this step. However, training_step() will be called whenever a new batch comes in. So training data numbers are 10% * batchsize * #ofbatches within an epoch.

My question is which one is the right RHO-LOSS? If the codebase is actually consistent with the paper, where can I find the "randomly select" part in it? Thank you very much and Merry Christmas!🎄

**Comment by mtrazzak** [comment]
Heya!

To answer the question in the subject: it selects points for a batch (not once an epoch). I'm not sure what the difference between the two explanations you have written down there: it is 10% of each batch is selected to be trained on. That's what is stated in Algorithm 1 and what is done in the codebase?

You can change to "randomly select" by setting selection_method in the config file to uniform_selection.yaml

Hope that clarifies things?

**Comment by xo28** [comment]
Hello Mtrazzak,

Thanks for your explanation! I used to think like this when I read the paper:
```
for epoch in epochs:
    randomly select a batch
    select 10% of this chosen batch
    train the model w/ this 10% of the chosen batch
```
But as you explained, it actually is:
```
for epoch in epochs:
    for batch in batches:
        select 10% of the current batch
        train the model w/ 10% of the current batch
```
So there are more data will be selected to train the model in each epoch. Additionally, I suppose both "randomly select" and "uniform_select" are methods of selecting 10% of a batch? 

Thank you!

**Comment by mtrazzak** [comment]
Sure!

Re: uniform/random: uniform sampling is a 'random' with uniform probability of every data point being chosen.
